internal/config/decode.go

```diff
@@ -90,9 +90,12 @@ func decodeString(
 
 	switch t {
 	case typeDuration:
-		// Check for simple integer values and treat them
+		// Check for simple integer value and if the value is positive treat it
 		// as milliseconds
 		if val, err := strconv.Atoi(dataString); err == nil {
+			if val < 0 {
```

Comment:
Now that I look at this, since this is Redis specific, I think it makes more sense to use a type alias for this in `internal/component/redis/settings.go`. It would just move the logic you have to the `DecodeString` function.

```golang
type Duration time.Duration

func (r *Duration) DecodeString(value string) error {
	if val, err := strconv.Atoi(value); err == nil {
		if val < 0 {
			*r = Duration(val)
			return nil
		}
		*r = Duration(time.Duration(val) * time.Millisecond)
		return nil
	}

	// Convert it by parsing
	d, err := time.ParseDuration(value)
	if err != nil {
		return err
	}

	*r = Duration(d)
	return nil
}
```

That of course creates a bunch of places where you need to cast/convert to `time.Duration` so I went ahead and attached a patch file.  Please review.

[patch.diff.zip](https://github.com/dapr/components-contrib/files/6542508/patch.diff.zip)

Reply:
@pkedy  LGTM! I've reviewed the changes, and as you said although `we don't know when a negative duration would make sense`, but processing the code in this way is more robust！

Reply:
I'm not sure which way you are (alias type vs globally) thinking is more robust. I'm going to assume it is OK to apply the patch to this branch. If not, we can revert.

Reply:
Sorry for didn't make myself clear. Briefly I prefer alias type too.